ghost/core/core/server/services/collections/BookshelfCollectionsRepository.js

```diff
@@ -122,19 +132,68 @@ module.exports = class BookshelfCollectionsRepository {
             {id: data.id},
             {
                 require: false,
-                transacting: options.transaction
+                transacting: options.transaction,
+                withRelated: ['collectionPosts']
             }
         );
 
         if (!existing) {
             await this.#model.add(data, {
                 transacting: options.transaction
             });
+            const thingsToInsert = collection.posts.map((postId, index) => {
+                return {
+                    id: (new ObjectID).toHexString(),
+                    sort_order: collection.type === 'manual' ? index : 0,
```

Comment:
Thinking out loud question, does the sort_order assignment type of logic belong to a repository or should we expect it to be handled on the entity level? I'm thinking the latter is more suitable, but then is sorting something entity should be even concerned about?

Reply:
`sort_order` is a persistence concern - it's just an implementation detail so that we can get the same entity back after saving!

Maybe an automatic collection should have a `Set` of post ids instead of `Array` so that it isn't implied that it should be ordered 🤔